src/host/readDataCooked.cpp

```diff
@@ -1049,11 +1046,14 @@ til::point COOKED_READ_DATA::_offsetPosition(til::point pos, ptrdiff_t distance)
 // It's intended to be used in combination with _writeChars.
 void COOKED_READ_DATA::_offsetCursorPosition(ptrdiff_t distance) const
 {
-    if (distance == 0)
+    if (distance != 0)
     {
-        return;
+        _offsetCursorPositionAlways(distance);
     }
+}
 
+void COOKED_READ_DATA::_offsetCursorPositionAlways(ptrdiff_t distance) const

```

Comment:
Would appreciate a method description here! Just to explain why this and `_offsetCursorPosition` can't just be combined into one function. The comment you added in `readDataCooked.cpp` explains it but if someone looks at this file on its own in the future it seems confusing.

Reply:
a better name might be `offsetCursorPositionAndResetState`?

Reply:
It doesn't reset state though? Maybe I'm misunderstanding you though...

Reply:
> ```
> // Using the *Always() variant ensures that we reset the blinking timer, etc., even if the cursor didn't move.
> ```

that sounds like resetting things to me!

Reply:
(but i understand!)